### PR TITLE
Migrate workflow engine to Flowable (#85).

### PR DESCRIPTION
**Pull request by karelmaxa**
This PR introduces the migration from the Activiti workflow engine to the Flowable engine. The migration is to version 6.7 because version 6.8 requires a newer OSGi specification and the new major version (7) is Java 17+ only. I expect the migration to the latest Flowable release to take place shortly after this PR.

One of the benefits of this change is that it allows us to upgrade other dependencies such as Groovy or Apache Felix.

The new version has been fully tested with the integration tests defined in the [wrenidm-test](https://github.com/orchitech/wrenidm-test) project.

**Breaking changes**

All process definitions must be migrated to use Flowable annotations. Migration instructions will be available later in the Wren:IDM Migration Guide.

### CODE REVIEW

**Comment by pavelhoral** [comment]
Please check if and how this should be covered by the semi-automated upgrade process (openidm-maintenance).

**Review by pavelhoral — CHANGES_REQUESTED** [review]

File: wrenidm-workflow-flowable/src/main/java/org/wrensecurity/wrenidm/workflow/flowable/WorkflowServiceImpl.java [hunk 1]
```diff
@@ -12,9 +12,9 @@
  * information: "Portions copyright [year] [name of copyright owner]".
  *
  * Copyright 2012-2016 ForgeRock AS.
- * Portions Copyright 2020 Wren Security
+ * Portions Copyright 2020-2024 Wren Security
  */
-package org.forgerock.openidm.workflow.activiti.impl;
+package org.wrensecurity.wrenidm.workflow.flowable;
```
[the remark this answers is not in the document]
Comment (karelmaxa) [reply wrenidm-workflow-flowable/src/main/java/org/wrensecurity/wrenidm/workflow/flowable/WorkflowServiceImpl.java]: Yes, I agree. The package `impl` has been taken back.

File: wrenidm-workflow-flowable/src/main/java/org/wrensecurity/wrenidm/workflow/flowable/WorkflowResource.java [hunk 2]
```diff
@@ -35,20 +35,24 @@
...
+import org.wrensecurity.wrenidm.workflow.flowable.resource.TaskInstanceHistoryResource;
+import org.wrensecurity.wrenidm.workflow.flowable.resource.TaskInstanceResource;
 
 /**
- * Implementation of the Activiti Engine Resource
- *
+ * Request handler providing workflow resources.
  */
-public class ActivitiResource implements RequestHandler {
+public class WorkflowResource implements RequestHandler {
```
[the remark this answers is not in the document]
Comment (karelmaxa) [reply wrenidm-workflow-flowable/src/main/java/org/wrensecurity/wrenidm/workflow/flowable/WorkflowResource.java]: OK, the class prefix has been changed to Flowable.

File: wrenidm-workflow-flowable/src/main/java/org/wrensecurity/wrenidm/workflow/flowable/WorkflowServiceImpl.java [hunk 3]
```diff
@@ -234,235 +223,207 @@ public Promise<ResourceResponse, ResourceException> handlePatch(Context context,
...
-                    processEngine = processEngineFactory.getObject();
-                    //We need to register the service because the Activiti-OSGi need this to deploy new BAR or BPMN
-                    Hashtable<String, String> prop = new Hashtable<String, String>();
-                    prop.put(Constants.SERVICE_PID, "org.forgerock.openidm.workflow.activiti.engine");
-                    prop.put("openidm.activiti.engine", "true");
-                    compContext.getBundleContext().registerService(ProcessEngine.class.getName(), processEngine, prop);
+                    // Register the OSGi service to enable deployment of BAR or BPMN files
+                    Dictionary<String, String> engineProperties = new Hashtable<>();
+                    engineProperties.put(Constants.SERVICE_PID, "org.forgerock.openidm.workflow.flowable.engine");
+                    engineProperties.put("wrenidm.flowable.engine", "true");
```
[the remark this answers is not in the document]
Comment (karelmaxa) [reply wrenidm-workflow-flowable/src/main/java/org/wrensecurity/wrenidm/workflow/flowable/WorkflowServiceImpl.java]: This is only used in this place, so in my opinion the constant is not necessary.

**Comment by karelmaxa** [comment]
As I wrote above, I will try to prepare migration notes including necessary changes in the IdM configuration and migration of the BPMN process definitions.

**Review by pavelhoral — APPROVED** [review]
LGTM

**Pull request merged**